numpy/_core/src/umath/dispatching.cpp

```diff
@@ -1358,6 +1359,67 @@ install_logical_ufunc_promoter(PyObject *ufunc)
     return PyUFunc_AddLoop((PyUFuncObject *)ufunc, info, 0);
 }
 
+NPY_NO_EXPORT int
+ldexp_promoter(PyObject *ufunc,
+        PyArray_DTypeMeta *op_dtypes[], PyArray_DTypeMeta *signature[],
+        PyArray_DTypeMeta *new_op_dtypes[])
+{
+    /* Promote integers (otherwise we do not match) to floating, i.e. double. */
+    new_op_dtypes[0] = PyArray_CommonDType(op_dtypes[0], &PyArray_FloatAbstractDType);
+    if (new_op_dtypes[0] == NULL) {
+        return -1;
+    }
+    new_op_dtypes[0] = ensure_concrete_dtype(new_op_dtypes[0]);
+    if (new_op_dtypes[0] == NULL) {
+        return -1;
+    }
+    /* Promote to at least int as we have int and int64 loops. */
+    new_op_dtypes[1] = PyArray_CommonDType(op_dtypes[1], &PyArray_IntDType);
+    if (new_op_dtypes[1] == NULL) {
+        Py_CLEAR(new_op_dtypes[0]);
+        return -1;
+    }
+    if (new_op_dtypes[2] == NULL) {
```

Comment:
Note to solve (or Maanas), forgot an INCREF.  Also not sure if we should check `signature`, although... I suppose it doesn't really matter, the promotion above should be always OK and if it causes a signature mismatch I think the later machinery would raise an error.

The test failures are all on platforms with `int` being 32bit integers, although I am not sure that is the underlying reason.
I somewhat suspect it may be an issue with the SIMD code, a way forward may to ignore this warning/FPE and check what the result type (and value, but if that fails just the type to see if the promotion is right).